config/src/main/scala/io/buoyant/config/Parser.scala

```diff
@@ -84,7 +84,7 @@ object Parser {
 
     // Subtypes must not conflict
     for (kinds <- configInitializers) {
-      ensureUniqueKinds(kinds)
+      kinds.groupBy(_.configClass.getSuperclass).values.foreach(ensureUniqueKinds)
```

Comment:
How do we feel about this change? Happy to add a test but wanted to run it by folks first. The previous code checked for uniqueness among all IDs for a given initializer class. The new code checks for uniqueness among all IDs for a given initializer class _and_ the parent config class.